src/WaferCircle.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { CSSProperties, FunctionComponent, useMemo } from 'react';
+
+interface WaferCircleProps {
+  diameter: number;
+  size: number;
+}
+
+function getCircleStyle(diameter: number): CSSProperties {
+  return {
+    height: diameter,
+    width: diameter,
+    position: 'relative',
+    borderRadius: '50%',
+    background: '#549ADA',
+  };
+}
+const WaferCircle: FunctionComponent<WaferCircleProps> = ({
+  diameter,
+  size,
+}) => {
+  const circleStyle = useMemo(() => getCircleStyle(diameter), [diameter]);
```

Comment:
You don't have to call `useMemo` so often. If the function that you call is fast, it may even be less performant.
Call `useMemo` either if the function is slow or if you want to prevent re-renders of children components (again, if the rerenders are slow).